libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -1002,6 +1002,7 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
             }
             // If fall through, there is a chance the package declares an inconsistent version, ignore it.
             isDevDepFallThrough = true;
+            break;
           }
 
           // eslint-disable-next-line no-fallthrough
```

Comment:
```suggestion
```

Reply:
Given that this explicit rule suppression is here, it seems like there was a reason for this behavior initially. Are you sure unit tests adequately cover all scenarios here?

Reply:
Or maybe I guess it was just an oversight.

Reply:
@chengcyber and I reviewed this today.  It is really a spec change, not a bug:

In the past, whether a package was a devDependency or regular dependency did not affect the `node_modules` folder. The distinction maybe was reflected in the lockfile, but the installation plan was equivalent.  

But now, apparently `rush install` fails if PNPM finds that the lockfile is inconsistent with **package.json** with regards to devDependencies-vs-dependencies.  We need to modify the spec, such that Rush's analysis now will treat `devDependencies` vs ` dependencies` as a significant difference.

This PR technically fixes the behavior by inserting a `break` statement, however the change is a bit confusing, because the code comments and structure are unchanged.  The /intent/ of the code appears to be the same, but in fact we have changed it.

Possible improvements:
1. Add some code comments better explaining the expected behavior
2. Possibly restructure the code to no longer rely on case statements that fallthrough
3. Ideally add some unit tests with examples of the edge cases